### PR TITLE
Rust updates, making VexRiscv tests error on missing binaries

### PR DESCRIPTION
**Pull request by cuddlefishie**

### CODE REVIEW

File: vex-riscv/vex-test-programs/src/bin/single_write.rs [hunk 1]
```diff
@@ -28,7 +28,7 @@ fn print(s: &str) {
 fn main() -> ! {
     let i = 14495943;
     let mut s = String::<16>::new();
-    let _ = writeln!(s, "Hey! {}", i);
+    let _ = writeln!(s, "Hey! {i}");
```
Comment (martijnbastiaan) [inline vex-riscv/vex-test-programs/src/bin/single_write.rs:31]: We need this in native Haskell..

**Review by martijnbastiaan — APPROVED** [review]
LGTM, thanks for tackling the warnings :). If you want to you can tack on making warnings errors on CI - I'll need to resolve merge conflicts in the Nix PR anyway.

Two nitpicks:

* Split `make VexRiscv integration tests error when binaries are missing, upda…` in two commits. Should make it easier to get the commit message within GitHub's limits too :-). 
* Start commit message with a capital letter

File: vex-riscv/vex-test-programs/src/bin/single_write.rs (hunk 1 above)
Comment (cuddlefishie) [reply vex-riscv/vex-test-programs/src/bin/single_write.rs:31]: Yea :pensive:

**Comment by cuddlefishie** [comment]
Thanks! I split the commit into two and updated the commit messages

**Review by martijnbastiaan — APPROVED** [review]

**Pull request merged**